impress-org/givewp
# Stripe can now be connected via the Setup Page

Resolves #4922

## Description


This implements the flow for connecting to Stripe via the Setup Page. The Stripe Connect button redirects to Stripe for authentication/authorization and handles the return request to complete the connection. Once connected the Stripe Connect action is no longer actionable via the Setup Page.

Additional changes:
- Updates the Stripe PHP SDK.
- Updates the "block-link" implementation to avoid conflict with a button link.
- Introduces a `RequestHandler` interface.
- No longer hides admin notices - I'd like to have a conversation about if we should do that.
- Extracts a new `PageView` class for separation of concerns.
- Refactors to leverage the new `Hooks` helper merged into `develop` by @JasonTheAdams 

NOTE: This does not cover the Success state of the Stripe Gateway row in the Setup Page, which requires additional design.

NOTE: This does not cover configuring webhooks for Stripe, which requires additional design.

## Affects



This modifies the check to connect Stripe via the Settings page, so that there isn't conflict with the Setup page flow.

## Visuals



## Pre-review Checklist




## Testing Instructions



From the Setup Page, click the Stripe Connect button to be redirected to the Stripe Connect flow. Once authenticated/authorized with Stripe, you should be redirected back to the Setup Page and the Stripe Connect button is no longer actionable.

## User Documentation
